code/code/misc/crit_combat.cc

```diff
@@ -559,6 +559,12 @@ int TBeing::critSuccessChance(TBeing *v, TThing *weapon, wearSlotT *part_hit, sp
   } else {
     double absdiff = abs(diff);
     double level_mod = 50.0 + ((double)diff * log(absdiff/20.0+1) / absdiff) * 75.0;
+
+    // Increasing scaling based on level difference for NPCs
+    if(dynamic_cast<TMonster *>(this)){
```

Comment:
Instead of dynamic_cast, we could have a method `virtual getSkillLevelMod()` for `TMonster`, with `TPlayer` overriding it.

Reply:
Before I attempt refactoring this - how big of an issue is this?  If it's a deal breaker I can give it a go, but looking at the file again it looks like dynamic_cast is called in this way a number of different times throughout the file to determine whether the individual in question is an NPC or a Player.  If it's not going to cause problems, I'd be inclined to suggest we leave it despite it being non-optimal.  If it may cause some issues, I can take a look into refactoring it though.  Let me know.

Reply:
It's not any worse than surrounding code, so I'm not demanding you fix it. Not that the surrounding code is anything beautiful to behold.
If you want to know if somebody is a player, it'd make sense to ask them if they are - have a isPlayer method in TBeing, with default return false, return true for TPerson.
Please note that there are situations where TBeings are also players - for example if an immort switches into a mob.